src/Microsoft.Android.Sdk.ILLink/FixAbstractMethodsHandler.cs

```diff
@@ -0,0 +1,293 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Mono.Cecil;
+
+using Java.Interop.Tools.Cecil;
+
+using Mono.Linker;
+using Mono.Linker.Steps;
+
+using Microsoft.Android.Sdk.ILLink;
+
+namespace Microsoft.Android.Sdk.ILLink
+{
+	public class FixAbstractMethodsHandler : IMarkHandler
```

Comment:
How different are `FixAbstractMethodsHandler` and `FixAbstractMethodsStep`? Would it be possible to keep a single `.cs` file and implement `IMarkHandler`? If so, could we continue sharing code with all the steps?

One reason this might be better is we run `FixAbstractMethodsStep` during `Debug` builds (without the linker):

https://github.com/xamarin/xamarin-android/blob/952e67be871b517208aa62ad6ae6679043669d8f/src/Xamarin.Android.Build.Tasks/Tasks/LinkAssembliesNoShrink.cs#L61

It could get confusing if a .NET 6 Debug build uses a different class for this than a Release build.

We were able to make `<LinkAssembliesNoShrink/>` fully incremental, so incremental builds it only runs on assemblies that change. We invoke some linker steps directly to make this work.

Reply:
Not that different. I changed it to use ifdefs, and most of the code is shared, but there are a few places where it impacts readability. The step can't implement both IMarkHandler and IStep, otherwise the linker doesn't know what kind of custom step this is.